app/validation/validators.py

```diff
@@ -212,31 +216,93 @@ def __call__(self, form, field):
             raise validators.ValidationError(self.message)
 
 
-class DateRangeCheck(object):
-    def __init__(self, messages=None):
-        if not messages:
-            messages = error_messages
-        self.messages = messages
+class SingleDatePeriodCheck(object):
+    def __init__(self, messages=None, minimum_date=None, maximum_date=None):
+        if messages:
+            self.messages = messages
+        else:
+            self.messages = error_messages
+        self.minimum_date = minimum_date
+        self.maximum_date = maximum_date
 
-    def __call__(self, form, from_field, to_field):
+    def __call__(self, form, field):
+        date_str = format_date_string(form)
+        date = convert_to_datetime(date_str)
+
+        if self.minimum_date:
+            if date < self.minimum_date:
+                raise validators.ValidationError(self.messages['SINGLE_DATE_PERIOD_TOO_EARLY'] %
+                                                 dict(min=self._format_playback_date(self.minimum_date + relativedelta(days=-1))))
```

Comment:
Same comment again about `relativedelta` - is it giving us anything that we don't already have with `datetime.timedelta`?

Reply:
answered above, actually looking at it again, it only deals with days, second, microseconds weirdly 😞